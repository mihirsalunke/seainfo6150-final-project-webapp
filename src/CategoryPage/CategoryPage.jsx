import React, { useEffect, useState} from "react";
import EventList from "../EventList/EventList.jsx";
import PropTypes from "prop-types";
import Loading from "../images/loading.gif";
import { isEmpty } from "lodash";

const CategoryPage = (props) => {

    const [eventsData, setEventsData] = useState();
    const [events, setEvents] = useState();

    useEffect(() => {
        const fetchEventsData = async () => {
          const eventsData = await fetch("http://demo0649072.mockable.io/events");
          const eventsJson = await eventsData.json();
          setEventsData(Object.values(eventsJson));
          let data = [];
          for(let i = 0; i < eventsJson.length; i++) {
              if(eventsJson[i].category === props.categoryName) {
                  data.push(eventsJson[i]);
              }
          }
          setEvents(data);
        };
    
        if (isEmpty(eventsData)) {
          fetchEventsData();
        }
    });

    return isEmpty(events) ? 
        <div   
            ><p>Fetching {props.categoryName} Events</p>
            <div>
                <img className="loading" src={Loading} alt="loading" />
            </div>
        </div> : (
        <>
            <div>
                <h1>{props.categoryName} Events:</h1>
            </div>
            <EventList events={events} />
        </>
    );
};

CategoryPage.propTypes = {
    categoryName: PropTypes.string.isRequired,
};

export default CategoryPage;