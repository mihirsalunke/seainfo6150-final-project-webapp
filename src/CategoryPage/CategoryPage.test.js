import React from "react";
import CategoryPage from "./CategoryPage.jsx";
import { render } from "@testing-library/react";

const categoryName = "Tech";

describe("CategoryPage tests", () => {
    it("renders correctly", () => {
      const { container } = render(<CategoryPage categoryName={categoryName} />);
      expect(container).toMatchSnapshot();
    });
});