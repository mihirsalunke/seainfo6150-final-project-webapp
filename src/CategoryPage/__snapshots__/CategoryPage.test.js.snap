// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPage tests renders correctly 1`] = `
<div>
  <div>
    <p>
      Fetching 
      Tech
       Events
    </p>
    <div>
      <img
        alt="loading"
        class="loading"
        src="loading.gif"
      />
    </div>
  </div>
</div>
`;
